dotCMS/src/main/java/com/dotmarketing/factories/TreeFactory.java

```diff
@@ -1,386 +1,363 @@
-					"' and tree.relationType = '" + relationType + "'");
-		} catch (DotHibernateException e) {
-		  throw new DotStateException(e);
-		}
-	}
-
-	public static void deleteTreesByParentAndChildAndRelationType(final String parent, final String child,
-																  final String relationType) {
-		try {
-
-			DotConnect dc = new DotConnect();
-			dc.setSQL("delete from tree where parent = ? and child = ? and tree.relation_type = ?");
-			dc.addParam(parent);
-			dc.addParam(child);
-			dc.addParam(relationType);
-			dc.loadResult();
-		} catch (DotDataException e) {
-			throw new DotStateException(e);
-		}
-	}
-
-
-	public static void deleteTreesByChildAndRelationType(Inode child, String relationType) {
-		try {
-			HibernateUtil.delete("from tree in class com.dotmarketing.beans.Tree where tree.child = '" + child.getInode() + 
-					"' and tree.relationType = '" + relationType + "'");
-		} catch (DotHibernateException e) {
-		  throw new DotStateException(e);
-		}
-	}
-	
-	public static void deleteTreesByChild(Inode child) {
-		try {
-			HibernateUtil.delete("from tree in class com.dotmarketing.beans.Tree where tree.child = '" + child.getInode()+"'");
-		} catch (DotHibernateException e) {
-		  throw new DotStateException(e);
-		}
-	}
-
-	public static void deleteTreesByChildId(final String childId) throws DotDataException {
-		DotConnect dc = new DotConnect();
-		dc.setSQL("DELETE FROM tree WHERE child = ?").addParam(childId).loadResult();
-	}
-
-	public static void deleteTreesByRelationType(String relationType) {
-		try {
-			HibernateUtil
-					.delete("from tree in class com.dotmarketing.beans.Tree where tree.relationType = '" + relationType + "'");
-		} catch (DotHibernateException e) {
-			throw new DotStateException(e);
-		}
-	}
-
-	public static void saveTree(Tree tree) {
-		try {
-			HibernateUtil.saveOrUpdate(tree);
-		} catch (DotHibernateException e) {
-		  throw new DotStateException(e);
-		}
-	}
-
-	
+
+
+    public static Tree getTree(final Inode parent, final Inode child) {
+
+        return getTree(parent.getInode(), child.getInode());
+    }
+
+    public static Tree getTree(final String parent, final String child) {
+        String relationType = "child";
+        return getTree(parent, child, relationType);
+
+    }
+
+    public static Tree getTree(final String parent, final String child, final String relationType) {
+        try {
+            DotConnect dc = new DotConnect();
+            String query = "select * from tree where parent = ? and child = ? ";
+            if (relationType != null)
+                query += " and relation_type = ?";
+            dc.setSQL(query);
+            dc.addParam(parent);
+            dc.addParam(child);
+            if (relationType != null)
+                dc.addParam(relationType);
+
+            return new DBTreeTransformer(dc.loadObjectResults()).tree();
+
+
+        } catch (DotDataException e) {
+            Logger.warn(TreeFactory.class, "getTree failed:" + e, e);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "getTree failed:" 6 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)